lib/utils/job-utils/catalog-searcher.js

```diff
@@ -50,8 +52,140 @@ function searchCatalogDataFactory(
+        }).filter(function (driveId) {
+            if (_.isEmpty(filter)) {
+                return true;
+            }
+
+            return driveId.identifier in filter || driveId.devName in filter;
+        }).then(function (driveIds) {
+            var vdCatalog, controllerCatalog;
+
+            _.forEach(driveIds, function (driveId) {
+                if (_.isEmpty(driveId.virtualDisk)) {
+                    return true; // continue
+                }
+                // find virtualDisk in megaraid catalogs
+                vdCatalog = waterline.catalogs.findMostRecent({
+                    node: nodeId,
+                    source: 'megaraid-virtual-disks'
+                });
+                // find OEMID from megaraid-controllers catalog
+                controllerCatalog = waterline.catalogs.findMostRecent({
+                    node: nodeId,
+                    source: 'megaraid-controllers'
+                });
+
+                return false; // break forEach
+            });
+
+            if(!vdCatalog) {
+                return driveIds;
+            }
+
+            return Promise.all([vdCatalog, controllerCatalog]).then(function (catalogs) {
+                if (!catalogs[0] || !_.has(catalogs[0], 'data.Controllers[0]')) {
+                    return Promise.reject(
+                        new Error('Could not find megaraid-virtual-disks catalog data.'));
+                }
+
+                virtualDiskData = _.get(catalogs[0], 'data');
+                controllerData = _.get(catalogs[1], 'data');
+                return driveIds;
+            });
+        })
+        .map(function (driveId) {
+            if (_.isEmpty(driveId.virtualDisk)) {
+                return driveId;
+            }
+
+            var match = driveId.virtualDisk.match(/^\/c(\d+)\/v(\d+)/);
+            if (!match) {
+                return driveId;
+            }
+
+            var vid = match[2];
+            var cid = match[1];
+            var vdInfo;
+            _.forEach(virtualDiskData.Controllers, function(controller, idx) {
+                var vd = _.get(
+                    controller,
+                    'Response Data[%s]'.format(driveId.virtualDisk)
+                );
+
+                if (!vd) {
+                    return true; // continue
+                }
+
+                vdInfo = vd[0];
+                vdInfo.pdList = _.get(controller,
+                    'Response Data[PDs for VD %d]'.format(vid)
+                );
+                vdInfo.oemId = _.get(controllerData,
+                    'Controllers[%d][Response Data][Scheduled Tasks].OEMID'.format(idx));
+                return false; // break forEach
+            });
+
+            if(!vdInfo) {
+                // clear virtualDisk if no matched info found in catalog
+                driveId.virtualDisk = '';
+                return driveId;
+            }
+            // set extended info to driveId catalog
+            driveId.size = vdInfo.Size;
```

Comment:
Sugget to add a controllerId as well. To identify a physical drives it is controllerId:enclosureId:slotId.